test/unit/gateways/adyen_test.rb

```diff
@@ -810,6 +810,16 @@ def test_authorize_with_network_tokenization_credit_card
     assert_success response
   end
 
+  def test_authorize_with_sub_merchant_id
```

Comment:
Does this test actually pass `sub_merchant_id`? :confused_england:

Reply:
Hah, whoops. It is not actually testing whether `sub_merchant_id` gets passed.